src/avram/form.cr

```diff
@@ -234,16 +234,25 @@ abstract class Avram::Form(T)
     _changes = {} of Symbol => String?
     database_fields.each do |field|
       if field.changed?
-        _changes[field.name] = if field.value.nil?
-                                 nil
-                               else
-                                 field.value.to_s
-                               end
+        _changes[field.name] = cast_value(field.value)
       end
     end
     _changes
   end
 
+  private def cast_value(value : Nil)
+    nil
+  end
+
+  private def cast_value(value : Object)
+    case value
+    when JSON::Any
+      value.to_json
+    else
```

Comment:
What else could this object be? I wonder if we should be explicit about what this object type is?

Reply:
This can be any object that the form supports. Before it was just doing

```
if field.value.nil?
  nil
else
  field.value.to_s
end
```

This just adds in the catch for `JSON::Any` to call `to_json` instead of `to_s`, but otherwise works exactly the same as before.

Reply:
Gotcha! Thanks for clarifying. It's been awhile since I wrote this lol